opennlp-tools/src/main/java/opennlp/tools/formats/irishsentencebank/IrishSentenceBankDocument.java

```diff
@@ -0,0 +1,262 @@
+    sentences = new ArrayList<IrishSentenceBankSentence>();
+  }
+
+  public void add(IrishSentenceBankSentence sent) {
+    this.sentences.add(sent);
+  }
+
+  public List<IrishSentenceBankSentence> getSentences() {
+    return Collections.unmodifiableList(sentences);
+  }
+
+  /**
+   * Helper to adjust the span of punctuation tokens: ignores spaces to the left of the string
+   * @param s the string to check
+   * @param start the offset of the start of the string
+   * @return the offset adjusted to ignore spaces to the left
+   */
+  private static int advanceLeft(String s, int start) {
+    int ret = start;
+    for (char c : s.toCharArray()) {
+      if (c == ' ') {
+        ret++;
+      } else {
+        return ret;
+      }
+    }
+    return ret;
+  }
+
+  /**
+   * Helper to adjust the span of punctuation tokens: ignores spaces to the right of the string
+   * @param s the string to check
+   * @param start the offset of the start of the string
+   * @return the offset of the end of the string, adjusted to ignore spaces to the right
+   */
+  private static int advanceRight(String s, int start) {
+    int end = s.length() - 1;
+    int ret = start + end + 1;
+    for (int i = end; i > 0; i--) {
+      if (s.charAt(i) == ' ') {
+        ret--;
+      } else {
+        return ret;
+      }
+    }
+    return ret;
+  }
+
+  public static IrishSentenceBankDocument parse(InputStream is) throws IOException {
+    IrishSentenceBankDocument document = new IrishSentenceBankDocument();
+
+    try {
+      DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+      DocumentBuilder docBuilder = docBuilderFactory.newDocumentBuilder();
+      Document doc = docBuilder.parse(is);
+
+      String root = doc.getDocumentElement().getNodeName();
+      if (!root.equalsIgnoreCase("sentences")) {
+        throw new IOException("Expected root node " + root);
+      }
+
+      NodeList nl = doc.getDocumentElement().getChildNodes();
+      for (int i = 0; i < nl.getLength(); i++) {
+        Node sentnode = nl.item(i);
+        if (sentnode.getNodeName().equals("sentence")) {
+          String src = sentnode.getAttributes().getNamedItem("source").getNodeValue();
+          String trans = "";
+          Map<Integer, String> toks = new HashMap<>();
+          Map<Integer, List<String>> flx = new HashMap<>();
+          List<Span> spans = new ArrayList<>();
+          NodeList sentnl = sentnode.getChildNodes();
+          int flexes = 1;
+          StringBuilder orig = new StringBuilder();
+
+          for (int j = 0; j < sentnl.getLength(); j++) {
+            final String name = sentnl.item(j).getNodeName();
+            if (name.equals("flex")) {
+              String slottmp = sentnl.item(j).getAttributes().getNamedItem("slot").getNodeValue();
+              Integer slot = Integer.parseInt(slottmp);
+              if (slot > flexes) {
+                flexes = slot;
+              }
+
+              if (flx.get(slot) == null) {
```

Comment:
replace this entire if stmt with  -  flx.computeIfAbsent(slot, k -> new ArrayList<>());